pkg/helpers/newapp/app/imageref.go

```diff
@@ -230,16 +230,27 @@ func (r *ImageRef) SuggestName() (string, bool) {
 	if r == nil {
 		return "", false
 	}
+	var name string
 	// if ImageStream already exists, its name should
 	// have higher preference than object name
-	if r.Stream != nil {
-		return r.Stream.Name, true
-	}
-	if len(r.ObjectName) > 0 {
-		return r.ObjectName, true
+	switch {
+	case r.Stream != nil:
+		name = r.Stream.Name
+	case len(r.ObjectName) > 0:
+		name = r.ObjectName
+	case len(r.Reference.Name) > 0:
+		name = r.Reference.Name
 	}
-	if len(r.Reference.Name) > 0 {
-		return r.Reference.Name, true
+	if len(name) > 0 {
+		if r.AsImageStream {
+			// in some certain situations, suggested ImageStream name might contain slashes
+			// in this case we want to replace it with dashes, referenced object might
+			// contain said dashes, but imagestream must cannot
+			// example: https://bugzilla.redhat.com/show_bug.cgi?id=1970805
+			name = strings.ReplaceAll(name, "/", "-")
+		}
```

Comment:
I think we could future-proof this some by eliminating the `r.AsImageStream` check and just always replacing slashes with dashes.
```suggestion

			// in some certain situations, suggested ImageStream name might contain slashes
			// in this case we want to replace it with dashes, referenced object might
			// contain said dashes, but imagestream must cannot
			// example: https://bugzilla.redhat.com/show_bug.cgi?id=1970805
			name = strings.ReplaceAll(name, "/", "-")
```

Reply:
At first I was going to argue that this is only the `ImageStream` issue, and other types of image references must be left intact, but after giving it a thought, I realized that you are right.
`SuggestedName` is used to generate names or part of names for internal `k8s` objects, and those names must follow the DNS standard, hence, no slashes. I think it is safe to assume we will never need slashes in any names here.

Do you think we might go even further and replace all the non[alphanumeric, dash or dot] symbols here?

Reply:
So, basically replace this with a regexp that replaces all non-valid characters with dashes? I think that sounds like a great idea.

Reply:
@coreydaley yeah, `[^0-9a-zA-Z-.]` should do the trick. From what I read, `k8s` names should go along with rfc1123, which references rfc952, which says, only alphanumeric, dots and dashes are allowed in a hostname. So, I thought it might be better to go with this more uniform approach rather than just slashes.